aten/src/ATen/native/TensorProperties.cpp

```diff
@@ -53,11 +53,37 @@ Tensor & detach_(Tensor & self) {
 }
 
 Tensor contiguous(const Tensor & self) {
-  if (self.is_contiguous()) {
-    return self;
-  }
-  return self.clone();
+  return contiguous(self, MemoryFormat::Contiguous);
 }
 
+Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
+  if (self.is_contiguous(memory_format)) {
+    return self;
+  }
+  auto result = at::empty_like(self);
+  switch (memory_format) {
+    case MemoryFormat::Any: // Back compatibility with old defaults
+    case MemoryFormat::Contiguous: {
+      break;
+    }
+    case MemoryFormat::ChannelsLast: {
+      AT_CHECK(
+          result.dim() == 4,
+          " required rank 4 tensor to use channels_last format");
```

Comment:
nit: Maybe it would be OK to suggest to users that we might relax this restriction in the future

Reply:
I want to 'relax' even before we officially put support into documentation. We just need to decide about transformations between NHWC -> NSC -> NHWC